src/components/v2/Icon/icons/checkboxBorder.tsx

```diff
@@ -0,0 +1,10 @@
+import * as React from 'react';
+import { SVGProps } from 'react';
+
+const SvgCheckboxBorder = (props: SVGProps<SVGSVGElement>) => (
+  <svg viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg" {...props}>
+    <rect x="0.5" y="0.5" width="23" height="23" rx="5.5" stroke="currentColor" />
+  </svg>
+);
```

Comment:
To continue on why I think we shouldn't use MUI; we wouldn't need to use an svg to set the border.
If we scale the checkbox, the SVG will scale with it and so the border width won't be constant.

Reply:
I don't see an issue here. Commonly you style a checkbox using :after and an adjacent element similar to this where it is using an `SVG` (also no uncommon). In either case we would be sizing the input and the styles

Reply:
Using an SVG is fine for the check icon, but as I mentioned using an SVG for the border means it won't scale properly if the checkbox width and height change.
This isn't an issue for now since we only use one size of checkbox, but it's common to have different checkbox size. If that time comes, we can make the update then.

Reply:
This component does scale. We can either use their provided size prop or we can use css. I pushed an example

Reply:
Yeah, so if you look at the border width of the scaled checkbox, it is thinner than the one of the "normal" sized checkbox: that was my point. If we ever have checkboxes which are smaller in the app, it is very likely designers will make them have the same border width.

As I said though, that's not the case now so I'm happy to approve the PR